src/main/webapp/app/exam/manage/exam-status.component.ts

```diff
@@ -0,0 +1,178 @@
+    PREPARE_EXERCISE_START = 'prepareExerciseStart',
+}
+
+export enum ExamReviewState {
+    UNSET = 'unset',
+    PLANNED = 'planned',
+    RUNNING = 'running',
+    FINISHED = 'finished',
+}
+
+export enum ExamConductionState {
+    PLANNED,
+    RUNNING,
+    FINISHED,
+}
+
+@Component({
+    selector: 'jhi-exam-status',
+    templateUrl: './exam-status.component.html',
+    styleUrls: ['./exam-status.component.scss'],
+})
+export class ExamStatusComponent implements OnChanges {
+    @Input()
+    public exam: Exam;
+
+    examConfigurationSteps = [
+        ExamConfigurationStep.CONFIGURE_EXERCISES,
+        ExamConfigurationStep.REGISTER_STUDENTS,
+        ExamConfigurationStep.GENERATE_STUDENT_EXAMS,
+        ExamConfigurationStep.PREPARE_EXERCISE_START,
+    ];
+    configurationStepToFlagMap = new Map<ExamConfigurationStep, boolean>();
+    examChecklist: ExamChecklist;
+    numberOfGeneratedStudentExams: number;
+
+    examPreparationFinished = false;
+    examConductionState: ExamConductionState;
+    examReviewState: ExamReviewState;
+
+    readonly examConfigurationStepEnum = ExamConfigurationStep;
+    readonly examConductionStateEnum = ExamConductionState;
+    readonly examReviewStateEnum = ExamReviewState;
+    readonly round = round;
+    readonly Math = Math;
+
+    // Icons
+    faTimes = faTimes;
+    faTimesCircle = faTimesCircle;
+    faCheckCircle = faCheckCircle;
+    faArrowRight = faArrowRight;
+    faDotCircle = faDotCircle;
+    faCalendarCheck = faCalendarCheck;
+    faCalendarTimes = faCalendarTimes;
+
+    constructor(private examChecklistService: ExamChecklistService) {}
+
+    ngOnChanges(): void {
+        // Step 1: Exam preparation
+        // Step 1.1: Configure exercises
+        this.configurationStepToFlagMap.set(ExamConfigurationStep.CONFIGURE_EXERCISES, this.areAllExercisesConfigured());
+        // Step 1.2: Register students
+        this.configurationStepToFlagMap.set(ExamConfigurationStep.REGISTER_STUDENTS, this.examChecklistService.checkAtLeastOneRegisteredStudent(this.exam));
+        this.examChecklistService.getExamStatistics(this.exam).subscribe((examStats) => {
+            // Step 1.3: Generate student exams
+            this.configurationStepToFlagMap.set(ExamConfigurationStep.GENERATE_STUDENT_EXAMS, this.examChecklistService.checkAllExamsGenerated(this.exam, examStats));
+            // Step 1.4: Prepare exercise start
+            const allExamsGenerated = this.configurationStepToFlagMap.get(ExamConfigurationStep.GENERATE_STUDENT_EXAMS)!;
+            this.configurationStepToFlagMap.set(ExamConfigurationStep.PREPARE_EXERCISE_START, !!examStats.allExamExercisesAllStudentsPrepared && allExamsGenerated);
+            this.examPreparationFinished = this.isExamPreparationFinished();
+            this.examChecklist = examStats;
+            this.numberOfGeneratedStudentExams = this.examChecklist.numberOfGeneratedStudentExams ?? 0;
+        });
```

Comment:
The other steps are updated in their own method, I think this one should be extracted to one as well.